src/rpc/fees.cpp

```diff
@@ -79,7 +79,10 @@ static RPCHelpMan estimatesmartfee()
             UniValue errors(UniValue::VARR);
             FeeCalculation feeCalc;
             CFeeRate feeRate{fee_estimator.estimateSmartFee(conf_target, &feeCalc, conservative)};
-            if (feeRate != CFeeRate(0)) {
+            // Return the maximum of minrelaytxfee, mempoolminfee and estimateSmartFee fee rate when
+            //  a) estimateSmartFee successfully recommended a fee rate
+            //  b) fee rate estimator has recorded enough blocks to be able to provide an estimate for conf_target.
+            if (feeRate != CFeeRate(0) || (unsigned int)feeCalc.returnedTarget >= conf_target) {
```

Comment:
Why require anything at all here? What's the difference really between seeing 1 tx in the last *n* blocks and seeing none? For the regtest user for example, the PR like this wouldn't actually solve the issue.